src/category_theory/limits/kan_extension.lean

```diff
@@ -0,0 +1,370 @@
+@[simp]
+def cone {F : S ⥤ D} {G : L ⥤ D} (x : L) (f : ι ⋙ G ⟶ F) :
+  cone (diagram ι F x) :=
+{ X := G.obj x,
+  π :=
+  { app := λ i, G.map i.hom ≫ f.app i.right,
+    naturality' := begin
+      rintro ⟨⟨il⟩,ir,i⟩ ⟨⟨jl⟩,jr,j⟩ ⟨⟨⟨fl⟩⟩,fr,ff⟩,
+      dsimp at *,
+      simp only [category.id_comp, category.assoc] at *,
+      rw [ff],
+      have := f.naturality,
+      tidy,
+    end } }
+
+variable (ι)
+
+/-- An auxiliary definition used to define `Ran`. -/
+@[simps]
+def obj_aux (F : S ⥤ D) [∀ x, has_limits_of_shape (index ι x) D] : L ⥤ D :=
+{ obj := λ x, limit (diagram ι F x),
+  map := λ x y f, limit.pre (diagram _ _ _) (index.map f),
+  map_id' := begin
+    intro l,
+    ext j,
+    simp only [category.id_comp, limit.pre_π],
+    congr' 1,
+    rw [index.map_id],
+  end,
+  map_comp' := begin
+    intros x y z f g,
+    ext j,
+    erw [limit.pre_pre, limit.pre_π, limit.pre_π],
+    congr' 1,
+    tidy,
+  end }
+
+/-- An auxiliary definition used to define `Ran` and `Ran.adjunction`. -/
+@[simps]
+def equiv [∀ x, has_limits_of_shape (index ι x) D] (F : S ⥤ D) (G : L ⥤ D) :
+  (G ⟶ obj_aux ι F) ≃ (ι ⋙ G ⟶ F) :=
+{ to_fun := λ f,
+  { app := λ x, f.app _ ≫ limit.π (diagram ι F (ι.obj x)) (index.mk (𝟙 _)),
+  naturality' := begin
+    intros x y ff,
+    simp only [functor.comp_map, nat_trans.naturality_assoc, obj_aux_map, category.assoc],
+    congr' 1,
+    erw [limit.pre_π, limit.w (diagram ι F _) (index.mk_hom (𝟙 _) ff)],
+    congr,
+    tidy,
+  end },
+  inv_fun := λ f,
+  { app := λ x, limit.lift (diagram ι F x) (cone _ f),
+    naturality' := begin
+      intros x y ff,
+      ext j,
+      erw [limit.lift_pre, limit.lift_π, category.assoc, limit.lift_π (cone _ f) j],
+      delta cone index.map,
+      tidy,
+    end },
+  left_inv := begin
+    intro x,
+    ext k j,
+    dsimp only [cone, diagram],
+    rw limit.lift_π,
+    simp only [nat_trans.naturality_assoc, obj_aux_map],
+    congr' 1,
+    erw limit.pre_π,
+    congr,
+    cases j,
+    tidy,
+  end,
+  right_inv := by tidy }
+
+/-- A variant of `Ran.equiv` with `whiskering_left` instead of functor composition. -/
+@[simps]
+def equiv' [∀ x, has_limits_of_shape (index ι x) D] (F : S ⥤ D) (G : L ⥤ D) :
+  (G ⟶ obj_aux ι F) ≃ (((whiskering_left _ _ _).obj ι).obj G ⟶ F) := equiv _ _ _
+
+end Ran
+
+/-- The right Kan extension of a functor. -/
+@[simps]
+def Ran [∀ X, has_limits_of_shape (Ran.index ι X) D] : (S ⥤ D) ⥤ L ⥤ D :=
+adjunction.right_adjoint_of_equiv (λ F G, (Ran.equiv' ι G F).symm) (by tidy)
+
+namespace Ran
+
+variable (D)
+
+/-- The adjunction associated to `Ran`. -/
+def adjunction [∀ X, has_limits_of_shape (Ran.index ι X) D] :
+  (whiskering_left _ _ D).obj ι ⊣ Ran ι :=
+adjunction.adjunction_of_equiv_right _ _
+
+end Ran
+
+namespace Lan
+
+local attribute [simp] comma.fst comma.map_right
+
+/-- The index category of limits used to define `Lan`. -/
+@[simp, derive category, nolint has_inhabited_instance]
+def index (l : L) := comma ι (functor.from_punit l)
```

Comment:
```suggestion
def index (l : L) := comma ι (functor.from_punit l)
```
Is there an advantage to making this a `@[simp] def` over an abbreviation? I feel like `simp` defs are generally sub-optimal

Reply:
No longer relevant, as I am now using (co)structured arrows.